lib/graphql/lang/ast/type_info.ex

```diff
@@ -58,7 +58,12 @@ defmodule GraphQL.Lang.AST.TypeInfo do
       name == Introspection.meta(:typename)[:name] ->
         Introspection.meta(:typename)
       parent_type.__struct__ == GraphQL.Type.ObjectType || parent_type.__struct__ == GraphQL.Type.Interface ->
-        parent_type.fields[name]
+        # FIXME: this "function or map" logic is repeated in the executor. DRY IT UP.
```

Comment:
This logic is repeated in two places. It seems like it should be on one of the GraphQL types (or a `Fields` protocol that's implemented by `ObjectType`, `Interface` and `Union`?)

@joshprice @markolson what do you think?